app/views/propertyrepresentation/appoint/appointAgentProperties.scala.html

```diff
@@ -230,17 +230,20 @@ <h2 class="govuk-heading-m">@messages("propertyRepresentation.search.filter")</h
                                 content = Text(messages("label.propertyRepresentation.appointedAgents"))
                             ),
                             items = SelectItem(
-                                        value = None,
-                                        text = if(organisationAgents.resultCount > 1 && agentAlreadyAppointed) {
+                                        value = params.agent,
+                                        text = if (params.agent.isDefined) {
```

Comment:
don't use `isDefined` followed by `.get` on Options
Basically what you want to do here is `getOrElse`

```scala
//extract the 3 ELSE clauses to a lazy val (lazy as they might not need to be computed at all
lazy val emptyAgentText = if (organisationAgents.resultCount > 1 && agentAlreadyAppointed) {
      messages("propertyRepresentation.appointAgentProperties.dropdown.multiple", organisationAgents.resultCount - 1)
  } else if (organisationAgents.resultCount > 1 && !agentAlreadyAppointed) {
      messages("propertyRepresentation.appointAgentProperties.dropdown.multiple", organisationAgents.resultCount)
  } else messages("propertyRepresentation.appointAgentProperties.dropdown.single")

// ...

// then use getOrElse
text = params.agent.getOrElse(emptyAgentText)
```

Reply:
Looks like a much more elegant way of handling this, I'll change it now, thanks.